src/Element.php

```diff
@@ -311,7 +311,7 @@ public function getLabelAttributes(): array
      *
      * Implementation will decide if this will overwrite or merge.
      *
-     * @return $this
+     * @return self
```

Comment:
The class is not final, so it needs `@return static`, 

see 

self for both parent and child: https://phpstan.org/r/e408ad6a-0bba-43c6-8d71-2005354de06c
static with self only on final: https://phpstan.org/r/5c73379c-5217-4cb4-bc84-a0ab4bbce09b

Reply:
@samsonasik 
I've reverted those specific changes from `$this` to `self`.

`$this` is the correct return type in all cases, but the psalm issues relating to `LessSpecificReturnStatement` just look _really_ odd to me. Can you take a look at the failures and see if you've got any ideas. I'm starting to feel a bit stumped, or there's possibly a psalm bug going on

Reply:
Can we try `static<TFilteredValues>`? Does it make sense?

Reply:
I have tried that with no success - also, we can't use that on `Fieldset::setUseAsBaseFieldset()` for example, because it doesn't have templates.

Adding a template to Fieldset etc might work for some of the issues, I could try that, but it's really not gonna work on `LabelAwareInterface` - if feels like some weird inheritance bug but I'm not clever enough to know that for sure.

Reply:
> Adding a template to Fieldset etc might work for some of the issues

This cannot work because `Fieldset` does not provide any way to access the filtered values after validation. `getValue` simply returns null.